converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Patient/PatientPipelineStep.cs

```diff
@@ -55,7 +64,23 @@ protected override async Task PrepareRequestImplementationAsync(FhirTransactionC
             throw new MissingRequiredDicomTagException(nameof(DicomTag.PatientID));
         }
 
-        var patientIdentifier = new Identifier(string.Empty, patientId);
+        //Patient system id is determined based on issuer id boolean
+        //If issuer id boolean is set to true, patient system id would be set to issuer of patient id (0010,0021)
+        //Other wise we will be using the patient system id configured during user provisioning
+        string patientSystemId = string.Empty;
+        if (_isIssuerIdUsed)
+        {
+            if (dataset.TryGetSingleValue(DicomTag.IssuerOfPatientID, out string systemId))
+            {
+                patientSystemId = systemId;
+            }
+        }
+        else
+        {
+            patientSystemId = _patientSystemId;
+        }
+        _logger.LogInformation("Patient system id :{PatientSystemId}", patientSystemId);
```

Comment:
I don't think we should log the value of a tag as it could be PII

Reply:
Removed the logging